twill-core/src/main/java/org/apache/twill/kafka/client/KafkaConsumer.java

```diff
@@ -33,14 +33,16 @@
 
     /**
      * Invoked when new messages is available.
+     * @param startOffset Offset of the first {@link FetchedMessage} in the iterator of new messages.
```

Comment:
Yes, the current implementation follows this at line 454. Is just the caller's responsibility to honor this?